### DISCUSSION TITLE
Co-assembly with rnaspades

### DISCUSSION THREAD

**Discussion by lilyyycao** [ablab/spades#1156] [Q&A] [answered]
Hello,

I currently have 18 FASTQ files that I would like to use to perform a co-assembly using rnaSPADES. I read the documentation but was still left unsure of how to go about inputting the 18 different files in the input parameter as if I read correctly there is a 9 file limitation. Any guidance on whether I need to reformat my 18 files in 1 file or any other suggestions will be appreciated, thank you.

**Comment by asl** [comment, marked as answer]
There are no limits of how many files you can specify for each library. So, if all your 18 datasets have the same sequencing characteristics (read length, library fragment length, etc.), then just specify all of the as a single library.

In any case, you can provide unlimited number of libraries via YAML dataset file

**Comment by ReemaSingh** [comment]
@asl What if the multiple FASTQ files (from the same species) are from two different projects with different library protocols? Can I still use them in a single rnaSPADES command to generate a single assembly?

**Reply by asl** [reply]
Nothing prevents you from supplying these files to rnaSPAdes. What will be the outcome is a separate question

**Reply by ReemaSingh** [reply]
Thank you for your response. From my previous test run - I generated a single assembly using 4 paired-end files (from two different Biorpojects) in rnaSPADES.  The command worked without any errors. And I am getting a "492387" transcript (with fairly good length) in the "hard_filtered_transcripts.fasta". However, after successful completion, I am getting a warning message 

"=== Error correction and assembling warnings:
 * 2:22:55.838    29G / 48G   WARN    General                 (pair_info_count.cpp       : 376)   Estimated mean insert size 162.871 is very small compared to read length 149

Is it acceptable if I ignore this warning message?

**Reply by asl** [reply]
It's up to you to decide. But as stated in the warning – your left reads almost fully overlap with the right ones. This might be expected, or may show some issues during the library preparation step

**Reply by ReemaSingh** [reply]
Okay, Got it. Thank you so much for your response.
